src/ex1/script.js

```diff
@@ -18,35 +18,50 @@ filterOption.onclick = filterTodo;
 function addTodo(event) {
     event.preventDefault();
 
-    const todoDiv = document.createElement("div");
-    todoDiv.classList.add("todo");
-
-    //create a new todo list item:
-    const newTodo = document.createElement("li");
-
     if (todoInput.value === "") {
         alert("ERROR - no empty task is allowed");
         return;
     }
+    const taskItem = createTodoDiv(todoInput.value)
+    todoList.appendChild(taskItem);
+    // clearing the input:
+    todoInput.value = "";
+}
+
+function createTodoDiv(todoInput) {
+    const newTodoDiv = document.createElement("div");
+    newTodoDiv.classList.add("todo");
+    newTodoDiv.appendChild(createTodoTitle(todoInput));
+    newTodoDiv.append(...createTwoButton());
+    return newTodoDiv
+}
 
-    newTodo.innerText = todoInput.value;
+function createTodoTitle(todoTitle) {
+    // this function create a new todo title and assign a class 
+    const newTodo = document.createElement("li");
+    newTodo.innerText = todoTitle
     newTodo.classList.add("todo-item");
-    todoDiv.appendChild(newTodo);
+    return newTodo
+}
 
-    //Check-mark button:
-    const completeButton = document.createElement("button");
-    completeButton.innerHTML = '<i class="fas fa-check"></i>';
-    completeButton.classList.add("complete-button");
-    todoDiv.appendChild(completeButton);
+function createTwoButton() {
+    // This function create 2 necessary buttons:
```

Comment:
relevant to all comments: when you add a comment think about the person who first reads this code. does he really need this comment to understand what the code does?

- if the answer is yes - it probably means the code is not readable enough and it may can be refactored to make it more readable (you will learn about it later on in the course).
- if the answer is no - it means that the reader reads the code twice - once when he reads the comment, and right after that when he reads the actual code.
- a good rule of thumb is to write comments only when you need to explain why you did that, and not what you did

you split the code into functions and your code is very readable, in this case we don't need those comments 🙂